src/main/java/org/folio/util/LogUtil.java

```diff
@@ -0,0 +1,136 @@
+import static java.util.function.UnaryOperator.identity;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.core.Response;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.rest.persist.PostgresClient;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpResponse;
+
+public class LogUtil {
+  private static final Logger log = LogManager.getLogger(LogUtil.class);
+  public static final String R_N_LINE_SEPARATOR = "[\\r\\n]";
+  public static final String R_LINE_SEPARATOR = "\\r";
+  private static final int MAX_OBJECT_JSON_LENGTH = 10 * 1024;
+  private static final int DEFAULT_NUM_OF_LIST_ELEMENTS_TO_LOG = 10;
+
+  private LogUtil() {
+    throw new IllegalStateException("Utility class");
+  }
+
+  public static String logAsJson(Object object) {
+    if (object == null) {
+      return null;
+    }
+
+    if (object instanceof JsonObject) {
+      try {
+        return crop(((JsonObject) object).encode());
+      } catch (Exception ex) {
+        log.warn("logAsJson:: Error while logging JsonObject", ex);
+        return null;
+      }
+    }
+
+    try {
+      return crop(PostgresClient.pojo2JsonObject(object).encode());
+    } catch (JsonProcessingException jsonProcessingException) {
+      log.warn("logAsJson:: Error while logging an object of type {}",
+        object.getClass().getCanonicalName(), jsonProcessingException);
+      return null;
+    } catch (Exception ex) {
+      log.warn("logAsJson:: Unexpected error while logging an object of type {}",
+        object.getClass().getCanonicalName(), ex);
+      return null;
+    }
+  }
+
+  public static String logList(List<?> list) {
+    return logList(list, DEFAULT_NUM_OF_LIST_ELEMENTS_TO_LOG);
+  }
+
+  public static String logList(List<?> list, int maxNumberOfElementsToLog) {
+    try {
+      if (list == null) {
+        return null;
+      } else {
+        int numberOfElementsToLog = min(list.size(), maxNumberOfElementsToLog);
+        return format("list(size: %d, %s: [%s])", list.size(),
+          numberOfElementsToLog == list.size() ? "elements"
+            : format("first %d element%s", numberOfElementsToLog, plural(numberOfElementsToLog)),
+          list.subList(0, numberOfElementsToLog).stream()
+            .map(LogUtil::logAsJson)
+            .collect(Collectors.joining(", ")));
+      }
+    } catch (Exception ex) {
+      log.warn("logList:: Failed to log a list", ex);
+      return null;
+    }
+  }
+
+  private static String plural(int number) {
+    return number == 1 ? "" : "s";
+  }
+
+  public static String logOkapiHeaders(Map<String, String> okapiHeaders) {
+    try {
+      return logAsJson(new JsonObject(okapiHeaders.keySet().stream()
```

Comment:
maybe use entrySet() here? Then  won't need to recalculate value for each key

Reply:
done